pkg/db/tx.go

```diff
@@ -128,18 +120,28 @@ func (d *dbTx) Exec(query string, args ...interface{}) (pgconn.CommandTag, error
 		"query": queryToString(query),
 		"took":  time.Since(start),
 	})
-	if isUniqueViolation(err) {
-		return nil, ErrAlreadyExists
-	}
 	if err != nil {
-		dbErrorsCounter.WithLabelValues("exec").Inc()
-		log.WithError(err).Error("SQL query failed with error")
-		return res, err
+		return res, d.handleSQLError(err, "exec", query)
 	}
 	log.Trace("SQL query executed successfully")
 	return res, err
 }
 
+func (d *dbTx) handleSQLError(err error, cmdType string, query string) error {
+	dbErrorsCounter.WithLabelValues(cmdType).Inc()
+	d.logger.WithError(err).Error("SQL query failed with error")
+
+	// Each error that is added here should be updated also in controller.go:handleAPIError
+	if isUniqueViolation(err) {
+		return ErrAlreadyExists
+	}
+	if pgxscan.NotFound(err) {
+		d.logger.Trace("SQL query returned no results")
+		return ErrNotFound
+	}
+	return fmt.Errorf("query %s: %w", query, err)
```

Comment:
Consider remove/change "query" to "%s statement: ..." or just "%s: ..." because the error can be from exec of insert operation and the word query can be confusing.